src/main/java/hudson/plugins/gradle/injection/GradleEnterpriseErrorsAction.java

```diff
@@ -0,0 +1,108 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.RootAction;
+import hudson.model.Run;
+import hudson.plugins.gradle.AbstractBuildScanAction;
+import hudson.util.RunList;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.StaplerProxy;
+
+import java.util.Date;
+import java.util.Iterator;
+
+@Extension
+public class GradleEnterpriseErrorsAction implements RootAction, StaplerProxy {
+    @Override
+    public String getIconFileName() {
+        return isVisible() && Jenkins.get().hasPermission(Jenkins.ADMINISTER) ? "/plugin/gradle/images/svgs/gradle-build-scan.svg" : null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Gradle Enterprise";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "gradle_enterprise";
+    }
+
+    @Override
+    public Object getTarget() {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        return isVisible() ? this : null;
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public Iterator<GeErrorModel> getErrors() {
+        return RunList.fromJobs((Iterable) Jenkins.get().allItems(Job.class))
+            .completedOnly()
+            .stream()
+            .filter(r -> {
+                AbstractBuildScanAction action = ((Run) r).getAction(AbstractBuildScanAction.class);
+                return action != null && (action.getHasGradleErrors() || action.getHasMavenErrors());
+            })
+            .limit(200)
+            .map(r -> GeErrorModel.fromRun((Run) r))
+            .iterator();
+    }
+
+    private boolean isVisible() {
+        return InjectionConfig.get().isCheckForBuildAgentErrors();
+    }
+
+    public static class GeErrorModel {
+        private final String buildTool;
+        private final String project;
+        private final String buildStatusIconClassName;
+        private final String buildStatus;
+        private final String buildUrl;
+        private final Date start;
+
+        private GeErrorModel(String buildTool, String project, String buildStatusIconClassName, String buildStatus,
+                            String buildUrl, Date start) {
+            this.buildTool = buildTool;
+            this.project = project;
+            this.buildStatusIconClassName = buildStatusIconClassName;
+            this.buildStatus = buildStatus;
+            this.buildUrl = buildUrl;
+            this.start = start;
+        }
+
+        static GeErrorModel fromRun(Run<?, ?> r) {
+            AbstractBuildScanAction action = r.getAction(AbstractBuildScanAction.class);
+            return new GeErrorModel(action.getHasGradleErrors() ? "GRADLE" : "MAVEN",
```

Comment:
We will be adding more build tools. It's better not to fall back to Gradle by default. 

Related note. At the moment the logic is split between 2 methods, `filter`, which checks for the error presence, and this method, which is assumed to be called only if the error exists. I think we can simplify it and do the check only here and return `Optional` (if no error or unsupported build tool) and have a filter just based on the returned `Optional`. WDYT?

Reply:
Yes, that sounds better, also I just realized that we need to support potentially multiple build tools errors per run, right ?